src/components/Header.tsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useRef } from 'react';
+
+type Props = {
+  onSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
+  title: string;
+  onTitleChange: (title: string) => void;
+  isAdding: boolean;
+};
+
+export const Header: React.FC<Props> = (props) => {
+  const {
+    onSubmit,
+    title,
+    onTitleChange,
+    isAdding,
+  } = props;
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, []);
+
+  return (
+    <header className="todoapp__header">
+      <button
+        data-cy="ToggleAllButton"
+        type="button"
+        className="todoapp__toggle-all active"
+        aria-label="Toggle All"
+      />
```

Comment:
Add the ability to toggle the completed status of all the todos with the `toggleAll` checkbox. 4 point in checklist